api/types.proto

```diff
@@ -30,12 +30,31 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message Set {
+	message Void {}
+	map<string, Void> val = 1;
+}
+
+message Integer {
+	uint64 val = 1;
+}
+
+message ThirdPartyResource {
```

Comment:
I have a feeling the name `ThirdPartyResource` may raise some concerns. We recently added a `Resource` type to our data store as a way to store generic objects inside the store (as opposed to swarm-native types like `Node`, `Task`, etc). `ThirdPartyResource` might be similar enough to this to be confusing.

Maybe `GenericNodeResource`? Not sure if anyone else has suggestions...